src/hooks/useDocsURL.ts

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'react';
+
+import { frameworkMap } from '../common/constants';
+import { Strings } from '../resources/Strings';
+import { Framework } from '../types/types';
+
+type SupportedPerfdocsFramework = 'talos' | 'awsy' | 'devtools';
+const supportedPerfdocsFrameworks: Record<SupportedPerfdocsFramework, string> =
+  {
+    talos: 'talos',
+    awsy: 'awsy',
+    devtools: 'performance-tests-overview',
+  };
+
+const useDocsURL = (suite: string, framework_id: Framework['id']) => {
+  const [docsURL, setdocsURL] = useState('');
+  const [isLinkSupported, setIsLinkSupported] = useState(true);
+
+  useEffect(() => {
+    const framework = frameworkMap[framework_id];
+    const supportedFramework =
+      supportedPerfdocsFrameworks[framework as SupportedPerfdocsFramework];
+    const urlReadySuite = suite.replace(/:|\s|\.|_/g, '-').toLowerCase();
+
+    if (framework_id === 12 && supportedFramework) {
```

Comment:
We shouldn't hardcode the id. For example here I don't know what framework has the id=12 and is not easy to follow.